### ISSUE TITLE
Refactoring: Prioritization of deserialization

### ISSUE THREAD

**Issue by mbtools** [larshp/abapGit#3536]
The order objects for deserialization is currently defined in ZCL_ABAPGIT_OBJECTS->PRIORITIZE_DESER and must be changed in case of new object dependencies.

Refactor so each object type can define itself, which other objects it depends upon. 

Suggestion:
```
ZIF_ABAPGIT_OBJECT->GET_DESERIALIZE_ORDER
  EXPORTING
    et_objects_before TYPE TABLE OF tadir-object
    et_objects_after    TYPE TABLE OF tadir-object 
```

Return list of objects that must be serialized before or after the current one.

**Comment by goreraks** [comment]
I dont think we need before and after output to decide deserialize order. I believe only et_objects_before[] is enough to decide the deserialize order.

**Comment by larshp** [comment]
```abap
  METHODS get_deserialize_order
    RETURNING
      VALUE(rt_objects_before) TYPE ty_items_tt.
```
